gnucash/register/register-gnome/completioncell-gnome.c

```diff
@@ -0,0 +1,1019 @@
+ * completioncell-gnome.c -- completion combobox cell for gnome     *
+ *                                                                  *
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+ *                                                                  *
+\********************************************************************/
+
+/*
+ * FILE: completioncell-gnome.c
+ *
+ * FUNCTION: Implement gnome portion of a entry completion combo widget
+ *           embedded in a table cell.
+ *
+ * HISTORY:
+ * @author Copyright (c) 2023 Robert Fewell
+ */
+
+#include <config.h>
+
+#include <string.h>
+#include <gdk/gdkkeysyms.h>
+
+#include "completioncell.h"
+#include "gnc-prefs.h"
+#include "gnucash-item-edit.h"
+#include "gnucash-item-list.h"
+#include "gnucash-sheet.h"
+#include "gnucash-sheetP.h"
+#include "table-allgui.h"
+#include "gnc-glib-utils.h"
+
+typedef struct _PopBox
+{
+    GnucashSheet* sheet;
+    GncItemEdit*  item_edit;
+    GncItemList*  item_list;
+
+    GHashTable*   item_hash; // the item hash table
+    GtkListStore* item_store; // the item list store
+
+    gchar*        newval; // string value to find
+
+    gboolean      signals_connected; // list signals connected
+    gboolean      list_popped;  // list is popped up
+
+    gboolean      autosize; // autosize the popup width
+
+    gboolean      sort_enabled; // sort of list store enabled
+    gboolean      register_is_reversed; // whether the register is reversed
+    gboolean      stop_searching; // set when there are no results
+
+    gboolean      strict; // text entry must be in the list
+    gboolean      in_list_select; // item selected in the list
+
+    gint          occurrence; // the position in the list
+    time64        time_now; // the current time
+    gboolean      add_separator; // need separator
+
+} PopBox;
+
+#define DONT_TEXT N_("Don't autocomplete")
+#define SEPARATOR 1073741824
```

Comment:
Please add a comment to explain why you took this number as `SEPARATOR` value. I suspect it is a certain time interval measured in seconds but that's not obvious to the casual (and future reviewer). If it's a time interval, it may be easier to comprehend to create it as a constant that shows the calculation rather than a macro:
```
const time64 x_years = x * 365 * 24 * 60 * 60;
```

Reply:
I is just a big number.